cmd/lifecycle/main.go

```diff
@@ -15,7 +16,7 @@ func main() {
 		cmd.Exit(err)
 	}
 
-	switch filepath.Base(os.Args[0]) {
+	switch strings.TrimSuffix(filepath.Base(os.Args[0]), ".exe") {
```

Comment:
I understand why we are doing this but it feels a bit odd to always trim `.exe` when it's only relevant for Windows. I wonder if a helper function could make this more clear.

Reply:
I'll be posting an update shortly that might happen to make this a tad nicer. Keep an 👁️  out for it :)